File: Selenium/Activities/Activity15.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity15 {
    public static void main(String[] args) {
        WebDriverManager.firefoxdriver().setup();

        // Suppressing browser log
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "NULL");
        // Initialize firefox driver
        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(20));

        // Open a page
        driver.get("https://training-support.net/selenium/dynamic-attributes");

        //Page title
        System.out.println("Title of page is: " + driver.getTitle());

        // Enter the credentials
        WebElement username = driver.findElement(By.xpath("//input[@placeholder='Username']"));
        username.sendKeys("admin");
        WebElement password = driver.findElement(By.xpath("//input[@placeholder='Password']"));
        password.sendKeys("password");
        WebElement login = driver.findElement(By.xpath("//button[@type='submit']"));
        login.click();

        // Confirm login msg
        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("action-confirmation")));
        WebElement loginMsg = driver.findElement(By.id("action-confirmation"));
        System.out.println("Loin message: " + loginMsg.getText());

        // close the browser
        driver.quit();
    }
}

File: Selenium/Activities/Activity15.py
from selenium import webdriver
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Open the link in browser
    driver.get("https://training-support.net/selenium/dynamic-attributes")

    wait = WebDriverWait(driver, 10)

    # Print page title
    print("Page Title is: " + driver.title)

    # Enter the credentials
    username = driver.find_element(By.XPATH, "//input[@placeholder='Username']")
    username.send_keys("admin")
    password = driver.find_element(By.XPATH, "//input[@placeholder='Password']")
    password.send_keys("password");
    login = driver.find_element(By.XPATH, "//button[@type='submit']")
    login.click()

    # Confirm login msg
    loginMsg = driver.find_element(By.ID, "action-confirmation")
    wait.until(expected_conditions.element_to_be_clickable(loginMsg))
    print("Loin message: " + loginMsg.text)

    # close the browser
    driver.quit()
